### ISSUE TITLE
Avoid problems by setting default network settings

### ISSUE THREAD

**Issue by bzub** [rook/rook#795]
I've seen three or four Rook users run into an issue where OSDs start trying to communicate with each other on the first IP of the Pod subnet of a node (example below). This could be a Ceph bug, it's been hard to diagnose so far as it pops up infrequently and is not yet reproducible.

It is a real time killer though when it does pop up, so that's why I'm proposing that we implement the workaround from [this issue](https://github.com/rook/rook/issues/605#issuecomment-297037276). It fixes the issue consistently and without introducing any known problems.

Essentially I'd like to add this to the default config for all daemons, or at least OSDs:
```
cluster network = $POD_CIDR
public network = $POD_CIDR
```

**Reference:**
#605 
https://gitter.im/rook/rook?at=5965e87089aea4761d8dd151
h/t: @matejc @Abdelsalam-Abbas

Log example of the issue:
```
2017-04-22 00:36:59.987364 I | osd0: 2017-04-22 00:36:59.987108 7f055ffbf700 -1 osd.0 10 heartbeat_check: no reply from 10.1.57.0:6803 osd.1 ever on either front or back, first ping sent 2017-04-22 00:36:39.878752 (cutoff 2017-04-22 00:36:39.987103)
2017-04-22 00:36:59.987406 I | osd0: 2017-04-22 00:36:59.987234 7f055ffbf700 -1 osd.0 10 heartbeat_check: no reply from 10.1.65.0:6803 osd.2 ever on either front or back, first ping sent 2017-04-22 00:36:39.878752 (cutoff 2017-04-22 00:36:39.987103)
2017-04-22 00:37:00.987590 I | osd0: 2017-04-22 00:37:00.987397 7f055ffbf700 -1 osd.0 10 heartbeat_check: no reply from 10.1.57.0:6803 osd.1 ever on either front or back, first ping sent 2017-04-22 00:36:39.878752 (cutoff 2017-04-22 00:36:40.987391)
2017-04-22 00:37:00.987625 I | osd0: 2017-04-22 00:37:00.987444 7f055ffbf700 -1 osd.0 10 heartbeat_check: no reply from 10.1.65.0:6803 osd.2 ever on either front or back, first ping sent 2017-04-22 00:36:39.878752 (cutoff 2017-04-22 00:36:40.987391)
2017-04-22 00:37:01.988050 I | osd0: 2017-04-22 00:37:01.987754 7f055ffbf700 -1 osd.0 10 heartbeat_check: no reply from 10.1.57.0:6803 osd.1 ever on either front or back, first ping sent 2017-04-22 00:36:39.878752 (cutoff 2017-04-22 00:36:41.987750)
```

**Comment by bassam** [comment]
ceph has logic to *pick* an ip address from a CIDR if the public/cluster ip address is not set explicitly. See https://github.com/rook/ceph/blob/rook-luminous/src/ceph_osd.cc#L445. For our scenarios, I don't see why we wouldn't unconditionally set `public_addr`, the new `public_bind_addr` and `cluster_addr` and avoid this whole picking addresses logic.

@bzub can you try setting `cluster_addr` explicitly and see if it fixes the problem.

@travisn @jbw976 this is a good fix to pickup for 0.5.

**Comment by bzub** [comment]
Unfortunately I don't have a way to reproduce the problem yet, and no clusters experiencing it currently. All I know after asking around is that it pops up after a node or mon is rebooted or lost. I'll post here if that changes though.

**Comment by bassam** [comment]
@Abdelsalam-Abbas can you please help us document steps to repro this one? We'd like to get this fixed for 0.5.

**Comment by Abdelsalam-Abbas** [comment]
@bassam AFAIK installing kubespray on azure using flannel would cause this issue. with default setting but only changing network plugin to flannel as kubespray has issues with calico and azure.

**Comment by bassam** [comment]
@Abdelsalam-Abbas great. I don't think we've used kubespray on azure before, do you mind listing out the steps?

**Comment by Abdelsalam-Abbas** [comment]
git clone https://github.com/kubernetes-incubator/kubespray.git
guide to setup azure variables :- 
https://github.com/kubernetes-incubator/kubespray/blob/master/docs/azure.md
then install ceph-common & `ceph-fs-common` on all nodes.
then deploy kubernetes using quick guide for kubernetes here.

@bassam If there is any unclear instruction, let me know

**Comment by bassam** [comment]
fixed in #806
